app.ts

```diff
@@ -0,0 +1,1160 @@
+		logger.info(`Mqtt api call: ${data.api}`)
+
+		let res: any, err: string
+
+		try {
+			switch (data.api) {
+				case 'updateNodeTopics':
+					res = gw.updateNodeTopics(data.args[0])
+					break
+				case 'removeNodeRetained':
+					res = gw.removeNodeRetained(data.args[0])
+					break
+				default:
+					err = `Unknown MQTT api ${data.apiName}`
+			}
+		} catch (error) {
+			logger.error('Error while calling MQTT api', error)
+			err = error.message
+		}
+
+		const result = {
+			success: !err,
+			message: err || 'Success MQTT api call',
+			result: res,
+			api: data.api,
+		}
+
+		socket.emit(socketEvents.api, result)
+	})
+
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises
+	socketManager.on(inboundEvents.hass, async function (socket, data) {
+		logger.info(`Hass api call: ${data.apiName}`)
+
+		let res: any, err: any
+		try {
+			switch (data.apiName) {
+				case 'delete':
+					res = gw.publishDiscovery(data.device, data.nodeId, {
+						deleteDevice: true,
+						forceUpdate: true,
+					})
+					break
+				case 'discover':
+					res = gw.publishDiscovery(data.device, data.nodeId, {
+						deleteDevice: false,
+						forceUpdate: true,
+					})
+					break
+				case 'rediscoverNode':
+					res = gw.rediscoverNode(data.nodeId)
+					break
+				case 'disableDiscovery':
+					res = gw.disableDiscovery(data.nodeId)
+					break
+				case 'update':
+					res = gw.zwave.updateDevice(data.device, data.nodeId)
+					break
+				case 'add':
+					res = gw.zwave.addDevice(data.device, data.nodeId)
+					break
+				case 'store':
+					res = await gw.zwave.storeDevices(
+						data.devices,
+						data.nodeId,
+						data.remove
+					)
+					break
+			}
+		} catch (error) {
+			logger.error('Error while calling HASS api', error)
+			err = error.message
+		}
+
+		const result = {
+			success: !err,
+			message: err || 'Success HASS api call',
+			result: res,
+			api: data.apiName,
+		}
+
+		socket.emit(socketEvents.api, result)
+	})
+}
+
+// ### APIs
+
+function isAuthEnabled() {
+	const settings = jsonStore.get(store.settings)
+	return settings.gateway && settings.gateway.authEnabled === true
+}
+
+async function parseJWT(req: Request) {
+	// if not authenticated check if he has a valid token
+	let token = req.headers['x-access-token'] || req.headers.authorization // Express headers are auto converted to lowercase
+	token = Array.isArray(token) ? token[0] : token
+	if (token && token.startsWith('Bearer ')) {
+		// Remove Bearer from string
+		token = token.slice(7, token.length)
+	}
+
+	// third-party cookies must be allowed in order to work
+	if (!token) {
+		throw Error('Invalid token header')
+	}
+	const decoded = await verifyJWT(token, sessionSecret)
+
+	// Successfully authenticated, token is valid and the user _id of its content
+	// is the same of the current session
+	const users = jsonStore.get(store.users)
+
+	const user = users.find(
+		(u: { username: any }) => u.username === decoded.username
```

Comment:
try to avoid inline function type declarations

Reply:
done